src/Extensions/Logging/Internal/Replayable/ReplayableActivityStopObserver.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions.Activities.Processing;
+
+namespace Microsoft.Omex.Extensions.Logging.Replayable
+{
+	internal class ReplayableActivityStopObserver : IActivityStopObserver
+	{
+		private readonly ILogEventReplayer m_logReplayer;
+		private readonly IOptions<OmexLoggingOptions> m_options;
+
+		public ReplayableActivityStopObserver(ILogEventReplayer logReplayer, IOptions<OmexLoggingOptions> options)
+		{
+			m_logReplayer = logReplayer;
+			m_options = options;
+		}
+
+		public void OnStop(Activity activity, object? payload)
+		{
+			if (m_options.Value.ReplayLogsInCaseOfError && ShouldReplayEvents(activity))
+			{
+				m_logReplayer.ReplayLogs(activity);
+			}
+		}
+
+
```

Comment:
Nitpick: Inconsistent use of single and double line breaks between methods in this class.

Reply:
> > ```
> >   	string applicationName,
> > ```
> 
> The parameters here are replicated throughout and then passed directly down the hierarchy of methods as far as I can tell, which is typically considered a code smell. It would be preferable to create a property bag class and pass this around, to avoid all of the replication
> 
> Refers to: src/Extensions/Logging/Internal/EventSource/OmexLogEventSource.cs:15 in [1f380b6](https://github.com/microsoft/Omex/commit/1f380b621457018f1f77775f38763a2bcd460432). [](commit_id = [1f380b6](https://github.com/microsoft/Omex/commit/1f380b621457018f1f77775f38763a2bcd460432), deletion_comment = False)

Unfortunately parameters signature is a part of `EvetSource` contract, it uses parameters (that need to be simple types) and parameter names to create an event.

Reply:
Not sure why my reply got inserted here, sorry, but empty lines also fixed, thank you